### DISCUSSION TITLE
speech bubble with arrows?

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#2090] [Ideas / Feature Requests]
MS Word supplies "revision", but I think it is not shown in a friendly way.

# 1. What I need in nicegui in short

At least, a component looks like a dialog bubble with arrow

- the arrow tip can be placed any place
- the font size, color of text in bubble can be changed
- the main shape of bubble is a rounded-corner rectangle, or rectangle. An oval one is not a good choice 

bonus features

- the color of text, bubble background color, bubble outline color can be changed 
- the arrow can be place on any direction of the bubble main shape
- the text in bubble can be wrapped automatically
- bubbles auto arrange to forbid overlapping 

<img src=https://cdn.vectorstock.com/i/1000x1000/07/39/speech-bubble-design-vector-18940739.webp width=20%/>

------
Following is the long story.

# 2. what I do in real human world

For example, if I read "helo world" on the paper and find an "l" is missing, I will use such sign. BTW, what will you use in your culture?
![on_paper](https://github.com/zauberzeug/nicegui/assets/7960913/4a49886c-a15c-4383-8028-045c095a2a1c)

# 3. let's go to the computer world

I remember that some UI supplies a tooltip with an arrow tail, just like these. I think I can use it to mimic what I do on the paper. Well I fake them in MS word by using "insert->shape":
![tooltip1](https://github.com/zauberzeug/nicegui/assets/7960913/1001f2cf-33ec-4a54-82f6-8076573ecd23)

But this kind of tooltip is not available in nicegui as far as I know:

- tooltip in nicegui is not a standalone component, on the other hand it must live with another component
- tooltip in nicegui only appears when mouse hover over its master. But I want to show it permanently
- the place of the arrow tip should be customized to any place

For easy reading, I prefer to this style, which is also made in MS word by using "insert->shape" 

![tooltip2](https://github.com/zauberzeug/nicegui/assets/7960913/36179b08-4f62-4ebf-bd84-c586620dd2ef)

For good looking, this arrow style is better which I draw after https://www.humaan.com/blog/creating-comic-book-speech-bubbles-with-svg-and-javascript/
![tooltip4](https://github.com/zauberzeug/nicegui/assets/7960913/ef96e315-96c0-453f-95c2-5ca6c3f7389a)

Thank you for reading.

**Comment by falkoschindler** [comment]
It's not exactly what you're looking for, but you _can_ insert tooltips like this:
```py
with ui.row():
    ui.label('Hello')
    ui.tooltip('foo').props('model-value no-parent-event :offset="[5,5]"')
    ui.label('World')
```
<img width="108" alt="Screenshot 2023-11-28 at 15 53 04" src="https://github.com/zauberzeug/nicegui/assets/5767091/31d2e0c6-7108-4ce6-9682-12a84327e1c2">

The little arrow tip could probably be added using some CSS magic.
But I can imagine you would like to insert the bubble anywhere within a text element, not only in between. This make things more complicated and I don't have a solution at hand.
